### PR TITLE
139 Added origin for api gateway in Cloudfront

### PR DESCRIPTION
**Pull request by mahinuzzaman**
Feat: #139 
- Added a new origin for CloudFront under the `/api` path, which routes to the API Gateway 
- Moved CloudFront in a separate module

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [x] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

Provide a brief overview of the changes introduced in this pull request.

## Related Issues/PRs (optional)

Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [ ] I have added or updated necessary documentation to reflect the changes.
- [ ] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [ ] I have checked for any potential conflicts with other branches or pull requests.
- [ ] I have updated necessary OpenAPI specs

## Reviewer Request (optional)

If you have any specific requests or suggestions for reviewers, mention them here.

## Screenshots (optional)

If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.

## Demo Link (optional)

If applicable, provide a link to a live demo or deployment where the changes can be reviewed.

## Known Issues (optional)

If there are any known issues or limitations with the changes made in this pull request, describe them here.

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [ ] No risk
- [x] Low
- [ ] High

### CODE REVIEW

File: iac/terraform/aws/auth/variables.tf [hunk 1]
```diff
@@ -2,3 +2,8 @@ variable "environment" {
   type        = string
   description = "Deployment environment"
 }
+
+variable "lambda_archive_path" {
+  type        = string
+  description = "Path to the Lambda function zip archives"
```
Comment (KhirulIslam) [inline iac/terraform/aws/auth/variables.tf:8]: which lambda's zip archive, that should be specified.

File: iac/terraform/aws/cloudfront/cloudfront.tf [hunk 2]
```diff
@@ -3,29 +3,46 @@ resource "aws_cloudfront_distribution" "cdn" {
...
   }
 
-
-  enabled             = true
-  is_ipv6_enabled     = false
-  comment             = "CloudFront distribution for front-end site"
-  default_root_object = "index.html"
+  # API Gateway Origin (for /api routes)
+  origin {
+    domain_name = "${var.rest_api_id}.execute-api.${data.aws_region.current.name}.amazonaws.com"
```
Comment (KhirulIslam) [inline iac/terraform/aws/cloudfront/cloudfront.tf:36]: can't we get this from api gateway reference ?

File: iac/terraform/aws/auth/variables.tf (hunk 1 above)
Comment (mahinuzzaman) [reply iac/terraform/aws/auth/variables.tf:8]: All lambda zips are generated in the same directory. so its value is same everywhere

File: iac/terraform/aws/cloudfront/cloudfront.tf (hunk 2 above)
Comment (mahinuzzaman) [reply iac/terraform/aws/cloudfront/cloudfront.tf:36]: yes we can do it, but the URL is like `https://st06ivxx1j.execute-api.ap-south-1.amazonaws.com`. we'll have to update the string to remove the https. so this way seemed cleaner

**Review by KhirulIslam — APPROVED** [review]

**Review by sakibstark11 — APPROVED** [review]

**Pull request merged**